crates/masonry/src/widget/button.rs

```diff
@@ -86,6 +86,9 @@ impl Widget for Button {
                     ctx.request_paint();
                     trace!("Button {:?} released", ctx.widget_id());
                 }
+                if ctx.is_active() && !ctx.is_hot() {
```

Comment:
The only scenario where this case would not happen (if the first "happy path" case didn't happen) is if:
1) The mouse button was down (and that didn't activate another widget); then
2) The cursor was dragged into this widget then released

I would probably remove this conditional entirely, and deal with the extra paint cycle in that case. That also lets us remove the `request_paint` from the first condition